crates/circuit/src/bit.rs

```diff
@@ -635,6 +652,12 @@ macro_rules! create_bit_object {
             fn get(&self, index: usize) -> Option<Self::Bit> {
                 self.0.get(index)
             }
+            fn is_owning(&self) -> bool {
+                match &*self.0 {
```

Comment:
Minor style suggestion: 
```suggestion
                match self.0.as_ref() {
```